microsoft/botframework-solutions
# Multiple Authentication Providers

## Description

This change adds support for multiple authentication providers in the virtual assistant and skills.

### Expected Behavior
- If user is not logged in to any provider, they will be prompted to add an account from the supported providers list
- If the user has already connected 1 account, that account will be used by default
- If the user has multiple accounts configured (only possible with Linked Accounts for now) they will be prompted to choose which of their accounts to user
- Supported providers are configured per skill

## Related Issue



 fixes #19 

## Testing Steps


1. Register a google app - instructions [here](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/google-logins?view=aspnetcore-2.1&tabs=aspnetcore2x) (instead of the Google+ API, enable the Google Calendar API)
2. Collect the clientId and clientSecret
3. Add the google auth provider in the Azure Bot Settings with the clientId and secret, and set the scopes to **https://www.googleapis.com/auth/calendar**
4. Add the appName, ClientId, ClientSecret, and scopes for your google app to the appsettings files
5. Say "logout" to ensure no tokens are cached 
6. Say "what's on my calendar"
7. See the auth provider prompt
![image](https://user-images.githubusercontent.com/27976702/48094854-1cf00180-e1c8-11e8-87c5-8cae57eb7814.png)
